app/tokyoto_app/app/controllers/top_controller.rb

```diff
@@ -9,6 +9,13 @@ def index
 
   def search
     @results = @query_fix.result
+    @results_distinct = @results.select(:support_id).distinct
+  end
+
+  def get_selected_condition_supports
+    @result_ids = params[:result_ids]
+    @selected_support_id = params[:selected_support_id]
+    @conditions_supports = @result_ids.map {|id| ConditionsSupport.find_by(id: id.to_i, support_id: @selected_support_id.to_i)}
```

Comment:
```suggestion
    @conditions_supports = (@result_ids.map {|id| ConditionsSupport.find_by(id: id.to_i, support_id: @selected_support_id.to_i)}).compact
```
以上にすれば、一旦エラーは出ないが、nilが入らないように二段階で実装した方と思います。